.github/workflows/ci.yaml

```diff
@@ -0,0 +1,79 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: [main, signal/gh]
+    tags: [v*.*.*]
+
+jobs:
+  build:
+    name: Test & Build
+    strategy:
+      matrix:
+        python-version: ['2.7', '3.7', '3.8', '3.10']
+    runs-on: [ubuntu-20.04]
+    env:
+      PYTHON: ${{ matrix.python-version }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # fetch all history for setuptools_scm to be able to read tags
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install python dependencies
+        run: |
+          pip install wheel build tox
+          pip install .
+      - name: Determine pyenv
+        id: pyenv
+        run: echo "value=py$(echo $PYTHON | tr -d '.')" >> $GITHUB_OUTPUT
+
+      - name: Run tests
+        env:
+          TOXENV: ${{ steps.pyenv.outputs.value }}
+        run: tox
+
+      - name: Build python package
+        run: python -m build
+
+      - name: Upload coverage
+        uses: codecov/codecov-action@v3
+        with:
+          env_vars: PYTHON
+          fail_ci_if_error: true
+          files: .coverage.${{ steps.pyenv.outputs.value }}.xml
+
+      - uses: actions/upload-artifact@v3
+        if: matrix.python-version == '2.7' || matrix.python-version == '3.8'
+        with:
+          name: dist-${{ matrix.python-version }}
+          path: dist
+
+  publish:
+    name: Publish to PyPI
+    needs: build
+    runs-on: [ubuntu-latest]
+    if: github.event_name != 'pull_request'
+    steps:
+      - uses: actions/download-artifact@v3
+
+      - name: Organize files for upload
+        run: |
+          mkdir dist
+          mv dist-3.8/* dist/
```

Comment:
Despite the name of the distribution folder being 3.8 (because it is built with the 3.8 interpreter), the resulting wheel should be compatible with any py3 version (`llbase-0.2.1.dev220-py3-none-any.whl`)